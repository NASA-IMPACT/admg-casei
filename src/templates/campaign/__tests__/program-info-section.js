import React from "react"
import { create } from "react-test-renderer"

import ProgramInfoSection from "../program-info-section"

const testString = "test string"
const testUrl = "https://www.test-campaign.io"
const testImage = {
  shortname: "ACES",
  logoAlt: "logo",
  category: "logo",
  logoImg: {
    childImageSharp: {
      fluid: {
        tracedSVG:
          "data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'",
        srcWebp:
          "/static/070840eb6299020cd65fd46e27c54400/36436/aces_logo.webp",
        srcSetWebp:
          "/static/070840eb6299020cd65fd46e27c54400/36436/aces_logo.webp 141w",
        srcSet:
          "/static/070840eb6299020cd65fd46e27c54400/29ce4/aces_logo.png 141w",
        sizes: "(max-width: 141px) 100vw, 141px",
        src: "/static/070840eb6299020cd65fd46e27c54400/29ce4/aces_logo.png",
        presentationHeight: 139,
        presentationWidth: 141,
        originalName: "aces_logo.png",
        originalImg:
          "/static/070840eb6299020cd65fd46e27c54400/29ce4/aces_logo.png",
        base64: "data:image/png;base64,iVBOR",
        aspectRatio: 1.014388489208633,
      },
    },
  },
}

describe("Program Info Section", () => {
  it("renders logo when present in props", () => {
    const component = create(
      <ProgramInfoSection
        id="program-info"
        logo={testImage}
        shortname="AirMOSS"
        fundingAgency={testString}
        fundingProgram={testString}
        programLead={testString}
        leadInvestigator={testString}
        dataManager={testString}
        repositoryWebsite={testUrl}
        partnerOrgListing={testString}
        publicationLink={testUrl}
      />
    )
    const tree = component.toJSON()
    const instance = component.root
    expect(instance.findAllByType("img")[0].props.alt).toBe(testImage.logoAlt)
    expect(tree).toMatchSnapshot()
  })
  it("renders placeholder when no logo is available", () => {
    const component = create(
      <ProgramInfoSection
        id="program-info"
        logo={{
          shortname: "GOES-R PLT",
          logoAlt: "GOES-R PLT logo",
          category: "logo",
          logoImg: null,
        }}
        shortname={testString}
        fundingAgency={testString}
        fundingProgram={testString}
        programLead={testString}
        leadInvestigator={testString}
        dataManager={testString}
        repositoryWebsite={testUrl}
        partnerOrgListing={testString}
        publicationLink={testUrl}
      />
    )
    const tree = component.toJSON()
    const instance = component.root
    expect(instance.findByType("img").props.src).toBeDefined()
    expect(tree).toMatchSnapshot()
  })
})
