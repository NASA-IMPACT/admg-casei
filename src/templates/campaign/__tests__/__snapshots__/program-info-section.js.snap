// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Program Info Section renders logo when present in props 1`] = `
<section
  className="inpage-nav"
  data-cy="program-info-section"
  id="program-info"
>
  <h2>
    Program Info
  </h2>
  <div
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#303641",
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": "0.618",
          "justifyContent": "center",
          "padding": "1rem",
        }
      }
    >
      <img
        alt="campaign-logo"
        data-cy="campaign-logo"
        src="https://via.placeholder.com/150"
      />
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "flex": "2.618",
          "gap": "1.5rem",
          "gridTemplateColumns": "minmax(0,1fr) minmax(0,1fr) minmax(0,1fr)",
          "padding": "2rem",
        }
      }
    >
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Funding Agency
        </label>
        <p
          data-cy="program-info-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Funding Program
        </label>
        <p
          data-cy="program-info-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Funding Program Lead
        </label>
        <p
          data-cy="program-info-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Lead Investigator
        </label>
        <p
          data-cy="program-info-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Data Manager / Technical Contact
        </label>
        <p
          data-cy="program-info-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Assigned Archive Repository
        </label>
        <a
          data-cy="program-info-content-link"
          href="test string"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#aac9ff",
            }
          }
          target="_blank"
        >
          test string
        </a>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Partner Organisation
        </label>
        <p
          data-cy="program-info-content-text"
        >
          N/A
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Tertiary Website
        </label>
        <a
          data-cy="program-info-content-link"
          href="test string"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#aac9ff",
            }
          }
          target="_blank"
        >
          test string
        </a>
      </div>
    </div>
  </div>
</section>
`;

exports[`Program Info Section renders placeholder when no logo is available 1`] = `
<section
  className="inpage-nav"
  data-cy="program-info-section"
  id="program-info"
>
  <h2>
    Program Info
  </h2>
  <div
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#303641",
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": "0.618",
          "justifyContent": "center",
          "padding": "1rem",
        }
      }
    >
      <img
        alt="campaign-logo"
        data-cy="campaign-logo"
        src="test-file-stub"
      />
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "flex": "2.618",
          "gap": "1.5rem",
          "gridTemplateColumns": "minmax(0,1fr) minmax(0,1fr) minmax(0,1fr)",
          "padding": "2rem",
        }
      }
    >
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Funding Agency
        </label>
        <p
          data-cy="program-info-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Funding Program
        </label>
        <p
          data-cy="program-info-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Funding Program Lead
        </label>
        <p
          data-cy="program-info-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Lead Investigator
        </label>
        <p
          data-cy="program-info-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Data Manager / Technical Contact
        </label>
        <p
          data-cy="program-info-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Assigned Archive Repository
        </label>
        <a
          data-cy="program-info-content-link"
          href="test string"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#aac9ff",
            }
          }
          target="_blank"
        >
          test string
        </a>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Partner Organisation
        </label>
        <p
          data-cy="program-info-content-text"
        >
          N/A
        </p>
      </div>
      <div
        data-cy="program-info-content"
      >
        <label
          data-cy="program-info-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "small",
              "textTransform": "uppercase",
            }
          }
        >
          Tertiary Website
        </label>
        <a
          data-cy="program-info-content-link"
          href="test string"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#aac9ff",
            }
          }
          target="_blank"
        >
          test string
        </a>
      </div>
    </div>
  </div>
</section>
`;
